### PR TITLE
⚠️(v3+ only)  feat: update cert-manager crd to v1.0 in kubebuilder v3

### PR DESCRIPTION
**Pull request by MartinForReal**
Signed-off-by: MartinForReal <fanshangxiang@gmail.cn>
Fixed: #1666 

Migrate cert-manager related CR to v1

### CODE REVIEW

File: pkg/plugin/v2/scaffolds/internal/templates/kustomize.go [hunk 1]
```diff
@@ -102,15 +102,15 @@ vars:
...
+#    version: v1
 #    name: serving-cert # this name should match the one in certificate.yaml
 #  fieldref:
 #    fieldpath: metadata.namespace
 #- name: CERTIFICATE_NAME
 #  objref:
 #    kind: Certificate
 #    group: cert-manager.io
-#    version: v1alpha2
+#    version: v1
```
Comment (camilamacedo86) [inline pkg/plugin/v2/scaffolds/internal/templates/kustomize.go:113]: The `pkg/plugin/v2` ought to receive only fix bugs and cannot have breaking changes. All improvements and etc needs to be addressed only  in `pkg/plugin/v3`. Could you please update it?

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
Really thank you for your contribution. 🥇  Just a few nits:

- Note that The pkg/plugin/v2 ought to receive only fix bugs and cannot have breaking changes. All improvements and etc needs to be addressed only in pkg/plugin/v3. Could you please update it?
- Also, note that it is not passing in the testdata. After change the templates we need to run `make generate` it will re-generate the testdata
- Could you please add (v3 only) in the title as well? 

PS.: the commits need to be squashed.

**Comment by MartinForReal** [comment]
I've changed the commit accordingly. FYI @camilamacedo86 Thanks!

**Comment by camilamacedo86** [comment]
Hi @MartinForReal, 

It shows great. See that it is not passing in the check to verify if the testdata/ samples are updated. See: https://travis-ci.org/github/kubernetes-sigs/kubebuilder/jobs/725451668#L1003-L1038

To update these samples we run `make generate`. Any change on the templates requires we run this command to do the pr. Could you please check it? Also, could you please add on the title (v3+ only)? And we need to start to use : 


```Contributors:

All code PR must be labeled with 🐛 (patch fixes), ✨ (backwards-compatible features), or ⚠️ (breaking changes)
Breaking changes will find their way into the next major release, other changes will go into an semi-immediate patch or minor release
For a quick PR template suggesting the right information, use one of these PR templates:
Breaking Changes/Features
Backwards-Compatible Features
Bug fixes
Documentation Changes
Test/Build/Other Changes
```

**Comment by MartinForReal** [comment]
Hi @camilamacedo86,

Thanks for your instructions~

File: testdata/project-v2-multigroup/config/certmanager/certificate.yaml [hunk 2]
```diff
@@ -1,16 +1,16 @@
 # The following manifests contain a self-signed issuer CR and a certificate CR.
 # More document can be found at https://docs.cert-manager.io
-# WARNING: Targets CertManager 0.11 check https://docs.cert-manager.io/en/latest/tasks/upgrading/index.html for 
+# WARNING: Targets CertManager 1.0.1 check https://docs.cert-manager.io/en/latest/tasks/upgrading/index.html for 
 # breaking changes
-apiVersion: cert-manager.io/v1alpha2
+apiVersion: cert-manager.io/v1
```
Comment (camilamacedo86) [inline testdata/project-v2-multigroup/config/certmanager/certificate.yaml:5]: v2 projects still been updated. It means that we are changing its template when we should not.

File: pkg/plugin/v2/scaffolds/internal/templates/kustomize.go (hunk 1 above)
Comment (camilamacedo86) [reply pkg/plugin/v2/scaffolds/internal/templates/kustomize.go:113]: See ^ we can just perform the change inside of `pkg/plugin/v3`. Could you please revert the changes on `pkg/plugin/v2` and re-run the make generate?

**Comment by camilamacedo86** [comment]
It shows ok now. Great work. Also, I checked in https://cert-manager.io/docs/installation/upgrading/upgrading-0.16-1.0/ and has not breaking changes that which would need to be addressed here.

File: testdata/project-v2-multigroup/config/certmanager/certificate.yaml (hunk 2 above)
Comment (camilamacedo86) [reply testdata/project-v2-multigroup/config/certmanager/certificate.yaml]: Solved. Feel free to close this one.

File: pkg/plugin/v2/scaffolds/internal/templates/kustomize.go (hunk 1 above)
Comment (camilamacedo86) [reply pkg/plugin/v2/scaffolds/internal/templates/kustomize.go]: Solved. Feel free to close this one.

File: test/e2e/v3/e2e_suite.go [hunk 3]
```diff
@@ -31,6 +31,9 @@ import (
 	"sigs.k8s.io/kubebuilder/test/e2e/utils"
 )
 
+const certmanagerVersion = "v1.0.1"
+const certmanagerURL = "https://github.com/jetstack/cert-manager/releases/download/%s/cert-manager-legacy.yaml"
+
```
Comment (camilamacedo86) [inline test/e2e/v3/e2e_suite.go:36]: it should not be added here. See; https://github.com/kubernetes-sigs/kubebuilder/blob/9406d91733acdf05cbb18d96bfe4733748af1202/test/e2e/utils/test_context.go#L30

We need to update it in the utils.

Comment (camilamacedo86) [reply test/e2e/v3/e2e_suite.go:36]: @MartinForReal ^

File: test/e2e/v3/e2e_suite.go [hunk 4]
```diff
@@ -31,6 +31,9 @@ import (
 	"sigs.k8s.io/kubebuilder/test/e2e/utils"
 )
 
+const certmanagerVersion = "v1.0.1"
+const certmanagerURL = "https://github.com/jetstack/cert-manager/releases/download/%s/cert-manager-legacy.yaml"
```
Comment (estroz) [inline test/e2e/v3/e2e_suite.go:35]: @MartinForReal you'll be able to customize which set of manifests to use (current vs legacy) with [kubectl version](https://github.com/kubernetes-sigs/kubebuilder/pull/1644/files#diff-5f5cf1a838c4d92e543721d2697ea0e2b0cf57b9f9a3c72bc9fa51a030bc56b8R80) once #1644 is merged.

File: test/e2e/utils/test_context.go [hunk 5]
```diff
@@ -27,9 +27,7 @@ import (
 	. "github.com/onsi/ginkgo" //nolint:golint
 )
 
-const certmanagerVersion = "v0.11.0"
 const prometheusOperatorVersion = "0.33"
-const certmanagerURL = "https://github.com/jetstack/cert-manager/releases/download/%s/cert-manager.yaml"
 const prometheusOperatorURL = "https://raw.githubusercontent.com/coreos/prometheus-operator/release-%s/bundle.yaml"
```
Comment (camilamacedo86) [inline test/e2e/utils/test_context.go:31]: We need to keep this one and then, in the e2e tests for the v3-alpha we should use the upper version and for the v2+ we should keep using this one.

**Pull request closed without merging**